include/glfw/window/window.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * -- GLFW thread safety key points --
+ *
+ * Initialization, termination, event processing and the creation and destruction of windows, cursors
+ * and OpenGL and OpenGL ES contexts are all restricted to the main thread.
+ *
+ * Because event processing must be performed on the main thread, all callbacks except for the error callback
+ * will only be called on that thread.
+ *
+ * All Vulkan related functions may be called from any thread.
+ *
+ */
+
+#pragma once
+
+#include <GLFW/glfw3.h>
+
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <span>
+#include <spdlog/spdlog.h>
+#include <string>
+
+#include "error.h"
+
+namespace wnd
+{
+
+class Window;
+class WindowManager;
+
+struct WindowConfig;
+
+using WindowType = GLFWwindow;
+using FramebufferSizeType = std::pair<int, int>;
+
+using WindowResizeCallbackSignature = void(
+    Window* window,
+    int width,
+    int height //
+);
+
+using ErrorCallbackSignature = void(
+    int error_code,
+    const char* description //
+);
+
+struct WindowConfig
+{
+    const int width = k_default_width;
+    const int height = k_default_height;
+    const char* title = k_default_title;
+    const bool fullscreen = k_default_fullscreen;
+
+    std::function<WindowResizeCallbackSignature> resize_callback = k_default_resize_callback;
+
+    static constexpr int k_default_width = 640;
+    static constexpr int k_default_height = 480;
+    static constexpr char k_default_title[] = "MinCraft";
+    static constexpr bool k_default_fullscreen = false;
+
+    static inline const std::function<WindowResizeCallbackSignature> k_default_resize_callback =
+        []( Window* window, int width, int height ) {
+            spdlog::warn( "No resize callback is set." );
+        };
```

Comment:
Why is this a lambda stored in std::function? This should be a static member function to avoid static member with storage.

Reply:
Because this is a plug and we won't really use it for our purposes. For not to pollute the source file, there is a lambda.